src/Parser/RawBody.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+
+class RawBody implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    protected $buffer = '';
+    protected $contentLength;
+
+    public function __construct(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->contentLength = $headers['content-length'];
```

Comment:
Honestly, I'd just skip body entirely when:
* `Content-Lenght` is `0`
* `Content-Length` is smaller than actual body

Every sane browser reports `Content-Length` correctly, so do sane programs. I found some specs quite flexible, but have doubts that 100% of the spec is absolutely necessary.

Also from the [spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html):
>  For compatibility with HTTP/1.0 applications, HTTP/1.1 requests containing a message-body MUST include a valid Content-Length header field unless the server is known to be HTTP/1.1 compliant. If a request contains a message-body and a Content-Length is not given, the server SHOULD respond with 400 (bad request) if it cannot determine the length of the message, or with 411 (length required) if it wishes to insist on receiving a valid Content-Length.

So 400/411 seems reasonable answer here.